{
  try {
    disconnect();
  }
 catch (  Exception e) {
    log.debug("Exception occurred during disconnecting from: " + getEndpoint() + " "+ e.getMessage());
  }
  super.doStop();
}
